### ISSUE TITLE
Allow anonymous shapes in Style rules

### ISSUE THREAD

**Issue by keenancrane** [penrose/penrose#985]
**Is your feature request related to a problem? Please describe.**
The Style language currently requires all shapes to be given an explicit name, even if these shapes never need to be referenced in subsequent expressions or rules.  For instance:

```
forall Node x {
   shape x.dotShape = Circle {
      center: (?,?)
      r: 10.0
      fillColor: Colors.black
   }
}
```

These "dummy names" (in this case, `x.dotShape`) make code less legible, and increase cognitive load.  It also forces the programmer to spend extra effort picking a dummy name that doesn't conflict with existing names, even though this name will never be used.

**Describe the solution you'd like**
A simple solution is to allow anonymous names, i.e., allow declarations like

```
forall Node x {
   shape Circle {
      center: (?,?)
      r: 10.0
      fillColor: Colors.black
   }
}
```

(Of course, if this shape does need to be referenced in later iterations of the program, it's easy enough to give it an explicit name.)

**Additional context**
This design nicely parallels the design we already have for constraints and objectives, which are allowed to be anonymous.  In fact, it sort of violates the principle of least surprise that constraints/objectives can be anonymous, but shapes cannot.

I would guess that anonymous shapes can be implemented exactly as we've done for anonymous constraints/objectives: automatically generate a name for each anonymous shape that only gets referenced internally within the compiler (but is never exposed to the user).

**Comment by ZiyaoWei** [comment]
I've been following Penrose for a while and super interested in getting involved - I know there was a "not ready for contribution" in the docs somewhere, but it seems to have gone. Would now be a good time? If so I'd be excited to start working on this issue!

**Comment by ZiyaoWei** [comment]
@wodeni - thanks for replying! That sounds great, I'll wait for it :-)

**Comment by ZiyaoWei** [comment]
@wodeni - since #984 has been merged I assume I can start working on this now?

**Comment by samestep** [comment]
@ZiyaoWei Thanks for reaching out again! Sorry for not following up earlier; @wodeni and I are both temporarily away from Penrose this summer due to internships, and I made a note to respond to you but hadn't gotten around to it yet.

Anyways, yes, that would be great! So, to clarify, after #984 I _expected_ this to already work (so I was going to instead point you to a different issue to work on), because now we treat all `AnonAssign` statements the same regardless of whether they're shapes or constraints/objectives/layerings/etc:

https://github.com/penrose/penrose/blob/880d197aa37290a43cccad9966f313f2d521b32f/packages/core/src/compiler/Style.ts#L1454

But just now @wodeni came up with an example using this, and it doesn't actually seem to work: https://penrose.cs.cmu.edu/try/?gist=80a8c2a48fc1cdf67e01be918cfba404

The same syntax error occurs whether or not you remove the `shape` keyword from the start of line 7 (I would have expected it to work without the `shape` keyword but not with it). So I'm guessing this issue will require changing the [Style parser](https://github.com/penrose/penrose/blob/880d197aa37290a43cccad9966f313f2d521b32f/packages/core/src/parser/Style.ne) but maybe not the Style compiler?

Let us know if you need any more guidance from our end.

**Comment by ZiyaoWei** [comment]
@samestep - thanks for the really detailed instructions, I'm sure that'll be super helpful. I'll take a look and enjoy you summer (and to @wodeni too)!

**Comment by ZiyaoWei** [comment]
Super long turnaround time, sorry :-)

I finally managed to get around to this, but if #1075 is correct this seems trivial?

OTOH I very much enjoyed going through the tutorial! I might have found a few bugs - will report/fix them. Thanks for the guidance above!

**Comment by samestep** [comment]
@ZiyaoWei

> Super long turnaround time, sorry :-)

No worries!

> I finally managed to get around to this, but if #1075 is correct this seems trivial?

Cool! Yeah, this looks like more or less what I expected. Thank you!

> OTOH I very much enjoyed going through the tutorial! I might have found a few bugs - will report/fix them. Thanks for the guidance above!

That sounds great! And if you're interested in contributing more, let me know if you'd like to point you to another issue you could work on.

**Comment by ZiyaoWei** [comment]
@samestep:

Thanks for the review and I am definitely interested in contributing more! Also thanks for the list and I will get started on them. Sorry for the slow reply, was out for a few days :-)
